### ISSUE TITLE
Re-design the "discover" view

### ISSUE THREAD

**Issue by mouse-reeve** [bookwyrm-social/bookwyrm#2751]
I think the view how I originally designed it has turned out to be visually messy and hard to use. I'd love for someone with a different perspective to re-think how it looks. 

The broader goal for this view is to expose users to what's going on on BookWyrm outside who they follow, but have an interface that guides users towards interacting differently with people outside their immediate network. For example, it's harder to reply to a post in the discover view, but easier to follow that user. My thinking here is that while on a technical level, a post may be public, it can still feel awkward or uncomfortable to get over-familiar replies from strangers, and while that's fundamentally more a social problem than a technical one, having the interface both look different and promote different types of interaction can help frame these content as contextually different than what's in your home feed.

If you'd like to work on this, you do not need to write any code at all if you don't want to -- providing a mock-up of an alternative layout for this view would be wonderful and I'd be happy to pick up wherever it ceases to be interesting for you.

**Comment by 85701225** [comment]
@mouse-reeve I agree this is an opportunity.  I don't think it has to be a substantial undertaking though - at least not all at once.

As an initial thought, shrinking the boxes to is-4 and adding buttons to engage with desired features makes a real dent in the problem and isn't a giant lift.

The mockup below is presented with the caveat that I'm not a UI/UX designer, but...  I think it captures the general solution at least: 3 per row instead of 2, with engagement opportunities.  I envision, but did not depict, the whitespace at the bottom left being a "reply" field.

![screenshot-bookwyrm social-2023 03 27-10_14_13](https://user-images.githubusercontent.com/103219771/227984737-c8c6f5fc-48e3-4cf0-b19e-3c85c9d10806.png)

I don't think this solves all the edge cases, but incremental improvement is a good approach to UI, and reduces the work in a given change - or in rolling it back, if something doesn't sit well with users.

**Comment by arjanboltjes** [comment]
As mentioned in #740, perhaps not just the visual side of "discover" may be in need of re-designing, but also the underlying filters. Right now, people seem to be discovered based on "followers", while for me it makes more sense if a parallel way of discovery would be based on types of books: this could be books you've both read or have on your shelves, genres you both read, subjects you're both interested in.

**Comment by 85701225** [comment]
@arjanboltjes Just saw #740, and I like the filtering improvements there.  Will give that some thought and mock something up.

**Comment by mouse-reeve** [comment]
@arjanboltjes We have the "books" timeline (on b.s, https://bookwyrm.social/books) which displays statuses about books on your shelves. I don't know that people are very aware of it

**Comment by MaggieFero** [comment]
@mouse-reeve The Books feed is nice, but it would be cool if people who shared more books read were suggested on the Discover pages, too. Right now, there's the possibility of getting stuck in a loop, where everybody you follow already follows each other. People reading similar books probably have similar interests to get into your feed, too, so they're sometimes a better source of suggestions than following-based recommendations (which get skewed by things like people following their actual friends and relatives whose book taste they don't share at all).

**Comment by hughrun** [comment]
Here's some suggestions for consideration, the first suggestion in particular may be controversial. I'm not particularly attached to any of them.

1. For book activity only include quotes and reviews - no readthrough activity or progress reports. This may not be such an issue if you implement item 2 below, but I feel like there's not a great deal of information being conveyed by a simple "started reading/finished reading".
2. Deduplicate activity by the same user on the same book. Currently if someone starts reading, posts a comment and then a progress post, the three things all appear next to or near each other in Discover. This looks kinda weird - I'd suggest only using the most recent activity, and if we exclude simple readthrough updates it may help or render this point moot.
3. Include a list of **new users** (who have opted in to be discoverable) to follow - potentially this could use an algorithm kind of the opposite of "people you might know" where it only shows users with no discernable links to you or your current follows/followers (if we're really trying to bust out of bubbles). Alternatively, it could just be the latest e.g. 10 users on your server.
4. **Latest X books added** to your local server
5. **Most popular books this week** in terms of activity
6. Blast from the past reads - random selection of books not read by anyone for X months
7. Top-rated books this week/month/year
8. Most controversial books with week/month/year - with a wide spread of 1 star and 5 star ratings

**Comment by arkhi** [comment]
Reading the existing comments, I would encourage suggestions not based on affinity or quantitative measurements, but on an option available for each user.

A proposal that could be setup like so:

* to _explore_ (suggest “out of my comfort zone” content);
* to _comfort_ (suggest “feel‑good” content).

Having the _explore_ by default would reduce the toxicity we see on other platforms, where the world tends to manichean behaviours. Giving the option to shelter oneself in times of crisis would show we care about each member of the community.

Having a _pause_ option where all notifications become quiet would also not be a bad thing. :) 

I hope this makes sense to others.

**Comment by ell1e** [comment]
I just thought this was the right place to bring this up:

> Recommendations from people, not algorithms

I saw this on the bookwyrm website and it feels like a warning, a downside. I get that if the algorithm dominates the feeds and platform **by default** that can be a bad thing if it's geared toward the wrong incentives, but that doesn't mean algorithms are bad. Without optional algorithms, discoverability can sometimes just be a lot poorer, since an algorithm can take into account more data than the friend connections often can.

Once this is adjusted, maybe this advertising line of "not algorithms" could maybe be changed to something like "Recommendations from people, not **just** algorithms".

**Comment by joric3** [comment]
Not sure if this should be a separate feature request (or two requests)?

I'd appreciate an option to filter my own activity out of what's shown to me on the "discover" page. Seeing my own activity again is not what I expect under a "discover" label - I expect things that are new to me. :) (I'm not talking about general visibility of my posts for others - just about what is shown to myself on this particular page.)
I'd be fine if the exclusion of a user's own activity would be general behavior of the discover page (instead of an option up to a user's choice).

I'd love to be able to filter for types of posts/activity to discover, e.g. reviews, comments, quotes, marking a book as "to read", started reading, finished reading, new user/profile, new book created/imported.

**Comment by joric3** [comment]
Still mulling over this and related questions. Based on what mouse-reeve writes above ("The broader goal for this view is to expose users to what's going on on BookWyrm outside who they follow"), another interesting filter option might not just be to include or exclude my own activity, but also in- or exclude all those who I already follow. Not sure what that would entail on a smaller instance, though. Discovering activity from those who follow me would also be nice.

More freewheeling thoughts:
Something like "show activity from users two nodes away from me" could be interesting. That is, e.g., someone who is followed by someone I follow. One step outside my immediate network, so to say.

Being able to easily discover user activity on other instances would be great in my opinion, but probably harder to implement? It would need some sort of suggestions algorithm, I assume.
